src/ripple/nodestore/ShardStoragePaths.md

```diff
@@ -0,0 +1,56 @@
+# Shard Storage Paths
+
+## Overview
+
+The shard database module stores validated ledgers in logical groups called shards. As of April 2020, a shard stores 16384 ledgers by default. In order to allow users to store shards on multiple devices, the shard database can be configured with several file system paths. While each path may refer to individual directories on a single device, the goal is to allow for paths stored on separate devices.
+
+## Configuration
+
+The `shard_db` and `historical_shard_paths` sections of the config file will be used to determine where the server stores shards. Minimally, the `shard_db` section must contain a single `main_storage_path` key. If this is the only storage path provided, all shards will be stored at this location. If the configuration also lists one or more lines in the `historical_shard_paths` section, all older shards will be stored at these locations, and the `main_storage_path` will be used only to store the current and previous shards. The goal is to allow users to provide an efficient SSD for storing recent shards, as these will be accessed more frequently, while using large mechanical drives for storing older shards that will be accessed less frequently.
+
+Below is a sample configuration snippet that provides a path for main storage and several paths for historical storage:
+
+```dosini
+# This is the persistent datastore for shards. It is important for the health
+# of the ripple network that rippled operators shard as much as practical.
+# NuDB requires SSD storage. Helpful information can be found here
+# https://ripple.com/build/history-sharding
+[shard_db]
+type=NuDB
+
+# A single path for storing
+# the current and previous
+# shards:
+# -------------------------
+main_storage_path=/var/lib/rippled/db/shards/nudb
+
+# Path where shards are stored
+# while being downloaded:
+# ----------------------------
+partial_shard_download_path=/var/lib/rippled/db/shards/
+
+# The number of historical shards to
+# store. The default value is 0, which
+# means that only the current and
+# previous shards will be stored.
+# ------------------------------------
+max_historical_shards=100
+
+# List of paths for storing older shards.
+[historical_shard_paths]
+/dev/sdb1
```

Comment:
I doubt that shards will be stored on raw devices? Might need a better path like `/mnt/disk1` for example.

Reply:
Done